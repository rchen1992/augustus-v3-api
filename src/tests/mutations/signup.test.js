const { testQuery } = require('@tests/utils');

const userName = 'newboi';
const password = 'password1';
const email = 'newboi@gmail.com';

testQuery({
    id: 'Sign up a new user',
    query: `
        mutation signup($input: SignupInput!) {
            signup(input: $input) {
                id
                userName
                email
                createdAt
            }
        }
    `,
    variables: {
        input: {
            userName,
            password,
            email,
        },
    },
    context: () => {
        return {
            services: {
                user: {
                    newUser() {
                        return {
                            user_id: 10,
                            user_name: userName,
                            email,
                            created_at: '2018-12-11 09:39:31',
                            updated_at: '2018-10-09 01:00:11',
                        };
                    },
                },
            },
        };
    },
    expected: undefined,
});
