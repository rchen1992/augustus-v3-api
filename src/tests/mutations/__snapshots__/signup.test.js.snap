// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Sign up a new user 1`] = `
Object {
  "data": Object {
    "signup": Object {
      "createdAt": "2018-12-11 09:39:31",
      "email": "newboi@gmail.com",
      "id": "10",
      "userName": "newboi",
    },
  },
}
`;
